README.md

```diff
@@ -1,338 +1,58 @@
-```json
-{
-  "autoload-dev": {
-    "files": [
-      "vendor/neighborhoods/throwable-diagnostic-component/test/Decorator/ThrowableDiagnosticMockerTrait.php",
-      "vendor/neighborhoods/throwable-diagnostic-component/test/Decorator/DiagnosedFactoryMockerTrait.php",
-      "vendor/neighborhoods/throwable-diagnostic-component/test/Decorator/DecoratorTestCase.php"
-    ],
-    "psr-4": {
-      "AcmeTest\\": [
-        "test/"
-      ]
-    }
-  }
-}
-```
-After changing `composer.json` run `composer dump-autoload` for the changes to have effect.
+Don't confuse releases on [packagist](https://packagist.org/packages/neighborhoods/throwable-diagnostic-component), e.g. 3.0.0, with versions of components contained within this package, e.g. AwsSqsV1. One release may contain multiple versions of the same component.  
+Parts of your code may use older versions of components, while others use the latest.
 
-The `DecoratorTestCase` is a [PHPUnit](https://github.com/sebastianbergmann/phpunit) `TestCase` using the `ThrowableDiagnosticMockerTrait` and `DiagnosedFactoryMockerTrait` traits. To make use of them PHPUnit must be a dev-dependency. A wide range of PHPUnit versions should be compatible with the autoloaded code.
+### Upgrade
 
-The test case for your custom decorator might look along these lines.
-```php
-<?php
+For upgrading from release 3 to release 4 follow the [Upgrade Guide](docs/UpgradeGuide.md).
 
-declare(strict_types=1);
-
-namespace AcmeTest\Unit\RiskyCode\ThrowableDiagnostic;
-
-use Acme\RiskyCode\ThrowableDiagnostic\Decorator;
-use Neighborhoods\ThrowableDiagnosticComponentTest\Decorator\DecoratorTestCase;
-use Throwable;
+## Usage
 
-class DecoratorTest extends DecoratorTestCase
-{
-    protected Decorator $decorator;
+Components fall into one of two categories
+* Throwable Diagnostic
+* Throwable Diagnostic Decorator
 
-    public function setUp(): void
-    {
-        parent::setUp();
+The Throwable Diagnostic entails the concept of how this works. The Throwable Diagnostic mostly contains interfaces. It determines how user code diagnoses a Throwable, how the decorators look like, how to define the stack of decorators to be used. You may think of it as a framework/skeleton/scaffolding.  
+There might be multiple versions of throwable diagnostic, e.g. ThrowableDiagnosticV1 and ThrowableDiagnosticV2.
 
-        $this->decorator = new Decorator();
-        $this->decorator
-            ->setDiagnosedFactory($this->getDiagnosedFactoryMock())
-            ->setThrowableDiagnostic($this->getThrowableDiagnosticMock());
-    }
+A Throwable Diagnostic Decorator is in charge to diagnose certain `Throwables`. For example the `GuzzleV1` component has a decorator which decides if an exception originating from guzzle is transient or not. During the diagnostic process a `Throwable` can be inspected by multiple decorators, each of which may decide that the `Throwable` is transient, permanent or should be passed to the next decorator.
```

Comment:
```suggestion
A Throwable Diagnostic Decorator is in charge of diagnosing certain `Throwables`. For example, the `GuzzleV1` component has a decorator which decides if an exception originating from guzzle is transient or not. During the diagnostic process a `Throwable` can be inspected by multiple decorators, each of which may decide that the `Throwable` is transient, permanent, or should be passed to the next decorator.
```

(that last comma is an "oxford comma". i prefer these, but they are definitely not required as a rule and i try not to suggest things that aren't necessary. however, here i feel like it helps clarify the meaning of the sentence.